source/postprocess/sea_level.cc

```diff
@@ -0,0 +1,584 @@
+                        scoord_new[2] = scoord[2];
+                        output_file << scoord_new << ' '<< nonuniform_sea_level_change << sea_level_offset << geoid_displacement << topography << ice_height << ocean_mask << std::endl;
+                      }
+                    if (nonuniform_sea_level_change > local_max_height)
+                      local_max_height = nonuniform_sea_level_change;
+                    if (nonuniform_sea_level_change < local_min_height)
+                      local_min_height = nonuniform_sea_level_change;
+                  }
+              }
+
+      // Calculate min/max non uniform sea_level change across all processes.
+      const double max_nonuniform_sea_level_change = Utilities::MPI::max(local_max_height, this->get_mpi_communicator());
+      const double min_nonuniform_sea_level_change = Utilities::MPI::min(local_min_height, this->get_mpi_communicator());
+
+      // Write results to statistics file.
+      statistics.add_value ("Minimum non-uniform sea_level change (m)",
+                            min_nonuniform_sea_level_change);
+      statistics.add_value ("Maximum non-uniform sea_level change (m)",
+                            max_nonuniform_sea_level_change);
+      const char *columns[] = { "Minimum non-uniform sea_level change (m)",
+                                "Maximum non-uniform sea_level change (m)"
+                              };
+      for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+        {
+          statistics.set_precision (columns[i], 8);
+          statistics.set_scientific (columns[i], true);
+        }
+
+      output_stats.precision(4);
+      output_stats << min_nonuniform_sea_level_change << " m, "
+                   << max_nonuniform_sea_level_change << " m";
+
+      // Write the solution to file.
+
+      // If this is the first time we get here, set the last output time
+      // to the current time - output_interval. This makes sure we
+      // always produce data during the first time step.
+      if (std::isnan(last_output_time))
+        {
+          last_output_time = this->get_time() - output_interval;
+        }
+
+      // Just return stats if text output is not required at all or not needed at this time.
+      if (!write_to_file || ((this->get_time() < last_output_time + output_interval)
+                             && (this->get_timestep_number() != 0)))
+        return std::pair<std::string, std::string> ("Non-uniform sea level change min/max:",
+                                                    output_stats.str());
+
+      std::string filename = this->get_output_directory() +
+                             "nonuniform_sea_level_change." +
+                             Utilities::int_to_string(this->get_timestep_number(), 5);
+      if (this->get_parameters().run_postprocessors_on_nonlinear_iterations)
+        filename.append("." + Utilities::int_to_string (this->get_nonlinear_iteration(), 4));
```

Comment:
I *think* (but haven't checked) that you only need the filename in the `if (Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0)` branch below. Move these lines there then.